### DISCUSSION TITLE
Attribution for annotation sets and individual annotations

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/annotations#6] [General]
A number of use cases documented in this draft hint at the concept of attribution:

> A teacher reads an ebook and prepares annotations. He exports an annotation file and shares this file with his students. The students import the ebook and the detached annotation file into their reading application: the annotations made by the teacher appear in the ebook. The students can add their annotations to the ebook, and these student annotations will not be mixed with the annotations created by their teacher.

> A user annotates an ebook. He shares his annotations with other users via a Cloud mechanism handled by his reading application.

There's very little information that relates to attribution/authorship in the current draft. I can see that the Annotation Set can contain a `dc:creator`, but I can imagine reading software putting their name into that field.

With cloud sync over a wide range of annotations/users, you'd also need to express this information per annotation rather on the entire set.

Is this purely meant to be handles through "layers" with the current draft? For example I could download an annotation set from a teacher and another one from the author and toggle them on/off at will?

While "layers" sound feasible to me (with additional properties and requirements), I'm wary of going down the road of full attribution/authorship, it's a very slippery one.

**Comment by HadrienGardeur** [comment]
As a side note, the current description for `dc:creator` says:

> The author, or comma separated authors.

Dealing with comma separated values is bad, especially with names where the name might be represented like this: "Last name, First name".

To support multiple values, it's best to repeat this property as many times as needed.

**Comment by llemeurfr** [comment]
The Annotation Set has a `generator`, not a `creator` (the creator property is in the About object; it is the author of the publication in Dublin Core speak). 

I agree that if the community decides that attribution is required, this should be done at the Annotation level, not the Annotation Set. I didn't see such requirement in the W3C Annotation model, which is a clue that it was not deemed as a requirement when [this spec](https://www.w3.org/TR/annotation-model/) was studied.  

> I'm wary of going down the road of full attribution/authorship, it's a very slippery one.
I agree.

> Is this purely meant to be handles through "layers" with the current draft? For example I could download an annotation set from a teacher and another one from the author and toggle them on/off at will?

Yes, this is a great benefit of the `tag` mechanism. 

> Dealing with comma separated values is bad, especially with names where the name might be represented like this: "Last name, First name".

I don't think it is bad in this case, as this data is only used as a helper for automatically associating an Association Set to a publication. Comparing string is much easier than comparing arrays. Let's not over-engineer things here.

**Reply by HadrienGardeur** [reply]
> Comparing string is much easier than comparing arrays. Let's not over-engineer things here.

The current draft already uses an array of strings for identifiers, what's the point of having a string separated value for `creator`?
We should avoid inconsistencies like that.

Comparing an array is hardly difficult and less likely to create tons of duplicated values if `creator` ever gets displayed to the user.
